src/modules/awake/AwakeModuleInterface/dllmain.cpp

```diff
@@ -69,26 +59,46 @@ class Awake : public PowertoyModuleIface
         unsigned long powertoys_pid = GetCurrentProcessId();
 
         std::wstring executable_args = L"--use-pt-config --pid " + std::to_wstring(powertoys_pid);
+        std::wstring application_path = L"modules\\Awake\\PowerToys.Awake.exe";
+        std::wstring full_command_path = application_path + L" " + executable_args.data();
         Logger::trace(L"PowerToys Awake launching with parameters: " + executable_args);
 
-        SHELLEXECUTEINFOW sei{ sizeof(sei) };
-        sei.fMask = { SEE_MASK_NOCLOSEPROCESS | SEE_MASK_FLAG_NO_UI };
-        sei.lpFile = L"modules\\Awake\\PowerToys.Awake.exe";
-        sei.nShow = SW_SHOWNORMAL;
-        sei.lpParameters = executable_args.data();
-        if (!ShellExecuteExW(&sei))
+        STARTUPINFO info = { sizeof(info) };
+
+        if (!CreateProcess(application_path.c_str(), full_command_path.data(), NULL, NULL, true, NULL, NULL, NULL, &info, &p_info))
```

Comment:
What is it the reason to use `CreateProcess` instead of `ShellExecuteExW`?

Reply:
`CreateProcess` gives me access to the process ID, that I can later use to query associated windows, and trigger the cleanup process for the tray. With `ShellExecuteExW` that information is not available.